src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1157,7 +1157,10 @@ protected override void ApplyDocumentInfoChanged(DocumentId documentId, Document
                 }
 
                 // Must save the document first for things like Breakpoints to be preserved.
-                projectItemForDocument.Save();
+                if (!projectItemForDocument.Saved)
+                {
+                    projectItemForDocument.Save();
+                }
```

Comment:
@jasonmalinowski are you familiar with this space at all?

Reply:
Not really. Chatting with @ryzngard privately it looks like this is an exception being thrown out of CPS, and thus is a CPS bug that it's throwing a null ref. @ryzngard is still investigating further.

Reply:
Yea, I filed https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1163405 to track it being fixed

Reply:
I do believe we can do this as a workaround for now

Reply:
So looking at the internal bug I think it's still not clear (at least to me) _why_ this works because we don't quite understand what's throwing in the first place. If the project system folks think this is good enough then I'm OK with it.

I _would_ however like it if we can update this code here to say it's a workaround and include the link to the bug so we can at least have some understanding of why it's here, and can perhaps remove it some day.

Reply:
Added comment and link to tracking bug